hudi-common/src/main/java/org/apache/hudi/common/util/ReflectionUtils.java

```diff
@@ -45,7 +45,7 @@
 
   private static final Logger LOG = LogManager.getLogger(ReflectionUtils.class);
 
-  private static final Map<String, Class<?>> CLAZZ_CACHE = new HashMap<>();
+  private static final Map<String, Class<?>> CLAZZ_CACHE = new ConcurrentHashMap<>();
```

Comment:
Thanks, can you explain why it is not thread-safe now, we already add a synchronized lock on the cache `CLAZZ_CACHE`.